src/rpc/misc.cpp

```diff
@@ -366,6 +368,32 @@ static UniValue setmocktime(const JSONRPCRequest& request)
     return NullUniValue;
 }
 
+static UniValue mockscheduler(const JSONRPCRequest& request)
+{
+        RPCHelpMan{"mockscheduler",
+            "\nBump the scheduler into the future (-regtest only)\n",
+            {
+                {"delta_time", RPCArg::Type::NUM, RPCArg::Optional::NO, "Number of seconds to forward the scheduler into the future." },
+            },
+            RPCResults{},
+            RPCExamples{""},
+        }.Check(request);
+
+    if (!Params().IsMockableChain()) {
+        throw std::runtime_error("mockscheduler is for regression testing (-regtest mode) only");
+    }
+
+    RPCTypeCheck(request.params, {UniValue::VNUM});
+    int64_t delta_seconds = request.params[0].get_int64();
+    if ((delta_seconds <= 0) || (delta_seconds > 3600)) {
+        throw std::runtime_error("delta_time must be between 0 and 3600 seconds (1 hr)");
+    }
+
+    g_rpc_node->scheduler->MockForward(boost::chrono::seconds(delta_seconds));
```

Comment:
in commit cf04e02bbe14257a1513b029231cb6a21c9766ac:

Would be nice to protect against null pointer dereference. See `EnsureMemPool`

Reply:
addressed in follow up